spec/datadog/appsec/waf_spec.rb

```diff
@@ -1348,9 +1336,9 @@
 
       expect(code).to eq :ddwaf_match
       expect(result[:timeout]).to eq false
-      expect(result[:data]).to_not be nil
-      expect(result[:actions]).to be_a Datadog::AppSec::WAF::LibDDWAF::ResultActions
-      expect(result[:actions][:size]).to eq 0
+      expect(result[:events]).to be_a Datadog::AppSec::WAF::LibDDWAF::Object
+      expect(result[:actions]).to be_a Datadog::AppSec::WAF::LibDDWAF::Object
+      expect(Datadog::AppSec::WAF::LibDDWAF.ddwaf_object_size(result[:actions])).to eq 0
```

Comment:
Minor: I wonder if it would make sense to convert the `LibDDWAF::Object`s into Ruby, and then assert on them as Ruby objects. (Similar note would apply to some of the other changed tests)

Reply:
We do that on other specs https://github.com/DataDog/libddwaf-rb/blob/17e014b63c3844b10616bfa3ff17fd2db634ee32/spec/datadog/appsec/waf_spec.rb#L1286-L1290. I believe these specs intend to test the internal objects when calling libddwaf directly using the libddwaf API. Loic wrote them, so I will ask him once he is available